extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -730,6 +773,35 @@ export class AmpA4A extends AMP.BaseElement {
     });
   }
 
+  renderViaSafeFrame_() {
+    user().assert(this.creativeBody_,
+        'creativeBody missing in renderViaSafeFrame_?');
+    this.lifecycleReporter_.sendPing('renderCrossDomainStart');
+    utf8Decode(this.creativeBody_).then(creative => {
+      /** @const {!Element} */
+      const iframe = this.element.ownerDocument.createElement('iframe');
+      iframe.setAttribute('height', this.element.getAttribute('height'));
```

Comment:
There's a decent amount of code duplication here with renderViaCrossDomainIframe_ (which btw is a bit of a misnomer... ideally we would rename it to renderViaClientNetworkCache_).  Let's create a helper function:

/**
 * Creates cross domain iframe to contain HTML/JS creatives whose src matches value given and initializes ApiHandler
 * @param {string} src
 * @param {opt_attributes=} {string, string} attributes
 */
void createCrossDomainIframe_(src, opt_attributes) {
    // create iframe
    // set default attributes
    // set optional attributes (e.g. name for safeframe)
    // initialize APIHandler
}